src/components/ChatInput/index.tsx

```diff
@@ -360,8 +371,13 @@ export const ChatInput: React.FC<ChatInputProps> = (props) => {
     if (variant !== "workspace") return;
 
     // Small delay to ensure DOM is ready and other components have settled
+    // Don't auto-focus if the terminal has focus (user is typing in terminal)
     const timer = setTimeout(() => {
-      focusMessageInput();
+      const activeElement = document.activeElement;
+      const terminalHasFocus = activeElement?.closest(".terminal-view") !== null;
+      if (!terminalHasFocus) {
+        focusMessageInput();
+      }
```

Comment:
I don't think this would trigger unless I'm confused. This would only happen on the initial render I believe unless something else is wack.